### PR TITLE
Convert timer functions to pybind11

### PR DESCRIPTION
**Pull request by sloretz**
part of #665

Converts:
* rclpy_reset_timer
* rclpy_call_timer
* rclpy_change_timer_period
* rclpy_is_timer_ready
* rclpy_cancel_timer
* rclpy_is_timer_canceled
* rclpy_time_until_next_call
* rclpy_time_since_last_call
* rclpy_get_timer_period
* rclpy_create_timer

### CODE REVIEW

**Comment by sloretz** [comment]
CI (build: `--packages-up-to rclpy` test: `--packages-select rclpy`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13791/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8661/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11502/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13872/) - I think there's currently a CI issue with Windows, so this one might fail and need to be rerun

File: rclpy/src/rclpy/timer.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  auto timer = std::unique_ptr<rcl_timer_t, decltype(deleter)>(
+    static_cast<rcl_timer_t *>(PyMem_Malloc(sizeof(rcl_timer_t))),
+    deleter);
+  if (!timer) {
+    throw std::bad_alloc();
+  }
+
+  *timer = rcl_get_zero_initialized_timer();
+
+  rcl_allocator_t allocator = rcl_get_default_allocator();
```
Comment (cottsay) [inline rclpy/src/rclpy/timer.cpp:223]: Should we create an `rcl_allocator_t` to use `PyMem_Malloc` (et al.)?

**Review by cottsay — APPROVED** [review]
One quick question, but otherwise LGTM.

File: rclpy/src/rclpy/timer.cpp (hunk 1 above)
Comment (sloretz) [reply rclpy/src/rclpy/timer.cpp:223]: I don't know. Profiling it sounds like a good idea, but I don't know what affect it would have on performance.

**Comment by sloretz** [comment]
CI after rebasing to fix merge conflict (build: `--packages-up-to rclpy` test: `--packges-select rclpy`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13817/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8681/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11527/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13903/)

**Comment by sloretz** [comment]
CI with hopefully no typo's in the args (build: ` --packages-up-to rclpy` test: `--packages-select rclpy`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13824/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8685/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11532/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13909/)

**Review by cottsay — APPROVED** [review]

**Pull request merged**